### ISSUE TITLE
[Feature]: Alternative/More original skills boosts

### ISSUE THREAD

**Issue by MarioSMB** [TeamLapen/Vampirism#1244]
### What?

The jump boost/strength skill choice is just a choice between two permanent potion effects. This overlaps with potions and beacons without providing a unique gameplay twist to vampires.  
Given the focus on movement powers, what if this skill instead passively allowed vampires to step up full blocks?

### Mod experience

I think this change would make for a better single selection that provides a powerful boost to vampire gameplay.
"Your speed and ferocity as a vampire allows you to scale entire blocks in a single stride"

**Comment by lunofe** [comment]
I find it problematic to pitch your preference as "the only logical choice".

Personally, I've been choosing Jump Boost since I started playing Vampirism, and I've even started putting Jump Boost beacons everywhere in my vanilla world _that's been around for 3 years now_, because it adds incredible value to me. I really don't want to play without it anymore.

In contrast, the ability to simply step up blocks without jumping is literally a known cheat and will conflict with Minecraft's "Auto Jump" accessibility option.

**Comment by MarioSMB** [comment]
Interesting, you might be the first person I've met who likes playing with the jump boost!  
In my experience and from what I've heard of others it tends to make navigation and traversal clunky, but there is a niche for everyone I suppose!  

While it is true that step-up is considered a cheat normally, it shouldn't cause conflicts with the auto jump option as step-up boots have been a part of mods since long before that option was added - it's a simple Forge property afterall.  
To have it as a unique ability for vampires would make for a powerful passive effect, and would not diminish the use of potions or beacons for those who want the jump boost effect.

EDIT: I reworded the description to clear up the bias towards jump boost - every buff has its uses!

**Comment by lusifer173** [comment]
Thats odd, I've been an active part of the vampirism community since 2019 and never heard anyone having a problem with jump boost, I and several people see it as a great ability that also really helps with movement.

It makes no sense to me to just go "you're fast and ferocious so you don't have to press space now" it also isn't a unique or powerful effect, its just not having to press space bar, infact that's pretty much already a feature in game with auto jump. Kinda just feels like one of those "everything must be vanilla+ ideas" although I may be wrong, thats just how I see it.

**Comment by MarioSMB** [comment]
The vanilla+ trend is just people copy-pasting what's already in the game, that exact logic is what I hope Vampirism can avoid by not using potion effects for its "special abilities".  

Auto-jump is mostly a hated feature - proof of that is the celebrations that were had when it was finally turned off by default.  
On the contrary, step-up is really powerful for movement and allows quick traversal of hills. It's also a mechanic that's not available in the vanilla game (not counting auto-jump) and is generally quite rare in mods, making it a good candidate for a passive ability.  
I'm all for unique effects though; if anything better can be found it would be preferable!

**Comment by Cheaterpaul** [comment]
I would be very careful about allowing vampires to auto step. It is a huge change and makes vampires agile than you may think. Autostepping practically means that the world looks like a flat world to your movement. While vampires are already faster than their counterpart, they will excel even more in hilly terrain.

You can already test this out with the survival form in the Werewolves addon mod. But there it limits the player to be in that specific form, while vampires would be just more powerfull.

**Comment by MarioSMB** [comment]
Indeed, it is a very powerful ability - maybe too much so for its current position, especially if werewolves already have it!  

Vampires already have access to flight and teleportation in the same tree, vertical agility may be excess on top of that... same could be said for jump boost though.  
It is worth trying and if it's found to upset the balance too much other ideas can be considered; for example, the additional armor and health you get just for levelling could be moved to a passive like this encouraging players to decide between more health or more resistance rather than which potion effect to be granted.

**Comment by Cheaterpaul** [comment]
The important difference between jumping and auto stepping is that jumping actively slows you down.

Yes, a jump boost gives you the ability to jump higher, but is slows you down and an attacker may use jump boosts or blocks to follow you.
Auto stepping allows the player to sprint up a mountain. With the additional speed of a vampire a pursuer has no chance at all to get to you. Yes, vampires can teleport and fly, but the bat form is very weak and both must be activated for use.

Where does the player choose between potion effects? The potions temporarily increase the attributes of the player. We simply increase the attributes directly, there is no need to use potion effects for things such as armor or health. It is naturally that a player gets more armor and health when leveling up, this is simply common practice and the way it works.

**Comment by MarioSMB** [comment]
By potion effects I am referring to one needing to choose between strength and jump boost (which are both normally provided by potions). I'm hoping to encourage a more creative take on skills by suggesting boosts that are not based on existing vanilla buffs.  

The stat increases from levelling up are quite drastic, which is why I would suggest making them a choice rather than passively granted for levelling - it is also common practice in games that you get to choose which attributes you want to pour your points into when levelling to fit your own playstyle and preferences.

You raise some very valid points about the power of step-up vs jump boost, for PvP it would indeed make vampires incredibly powerful.  
I can see how it may mess up the balance of fighting vampires - while they should definitely have quick getaway tools, passively granted ones may be too strong.  

\
On that note, feel free to close this issue; I am glad to know the ability exists in the Werewolves mod and will continue to ponder potential replacements for the potion effect skills. Thank you!

**Comment by maxanier** [comment]
I do agree that just using vanilla buffs as skills is a bit boring. Sometime it is just too convenient to use them instead of carefully implementing an original idea. Especially, because you need some original idea first. Therefore, I appreciate your suggestions. 
From these discussion I take that step-up should not be implemented, though. 
Moving the stat buffs from passively granted to actively chosen is an interesting option. While not too original an idea either, it would provide more choice and flexibility. We will definitely stay with these passive stat increase per level, that is just the style we have chosen for Vampirism (regardless of other games doing it differently), but we may remove some percentages there and re-add them as skill. (I'm not entirely sure, but I do believe this is what happened with the jump boost. In very early versions it was granted by default).

**Comment by kaboom24** [comment]
Stepping up blocks is already added by the vampiric aging Add-on by @TheDrOfDoctoring

**Comment by MarioSMB** [comment]
Perhaps the answer to this is to use attributes in place of potion effects where possible, not just for health and armor?  
For example, instead of granting the strength potion effect passively, the player could be given a 20% increase to their base attack damage (or attack speed if one wishes to balance it a bit differently). That way potions would still have their uses and there'd be a much wider array of stats to play with.

**Comment by Cheaterpaul** [comment]
Yes, direct attribute modification is preferred, but not for everything. For active effects it might sometimes to be better to use potion effects because they can not stack.
